Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.DataProtection.Storage/RetryHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2019 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading;
+
+namespace Google.Cloud.AspNetCore.DataProtection.Storage
+{
+    /// <summary>
+    /// Simple means of executing retry with exponential backoff and proportional jitter,
+    /// retrying on any GoogleApiException.
+    /// </summary>
+    internal sealed class RetryHandler
```

Comment:
Don't we already have code that does this??

Reply:
Sort of - but in the Gax.Grpc package, which we're not using here :( And I don't want to end up taking a transitive dependency on it for that reason.